engine/combat/combat_controller.py

```diff
@@ -0,0 +1,79 @@
+from control import sos_ctrl
+from engine.combat.utility.sos_reasoner import SoSReasoner
+from memory.combat_manager import CombatCharacter, combat_manager_handle
+
+combat_manager = combat_manager_handle()
+
+
+class CombatController:
+    def __init__(self):
+        self.reasoner = None
+        self.action = None
+        self.ctrl = sos_ctrl()
+
+    # returns a bool to feed to the sequencer
+    def execute_combat(self) -> bool:
+        self.ctrl.set_neutral()
+
+        # if combat is done, just exit
+        if combat_manager.encounter_done is True:
+            return False
+
+        # if reasoner doesn't exist, add it.
+        if combat_manager.encounter_done is False and self.reasoner is None:
+            self.reasoner = SoSReasoner(combat_manager)
+
+        # if we dont have an action or the current appraisal is complete,
+        # we make a new one.
+        # we also check if battle command has focus, so it doesn't start executing before
+        # we have control
+        if (
+            (self.action is None or self.action.appraisal.complete)
+            and combat_manager.selected_character is not CombatCharacter.NONE
+            and combat_manager.battle_command_has_focus
+        ):
+            # print("No action exists, executing one one")
+            self.action = self.reasoner.execute()
+            return False
+
+        # For some reason the action isn't set, so bail out.
+        if self.action is None:
+            # print("baling out because self action is nil")
+            return False
+
+        # if the consideration doesn't believe the situation is valid, execute it.
+        # This will put the cursor on the character it should be on.
+        # internally it checks to see if the character is not NONE and if the selected
+        # character is the one it wants and return true if so.
+        # if the character is None, it knows it can't move things.
+        consideration_valid = self.action.consideration.valid(
+            combat_manager.selected_character, self.action
+        )
+        if not consideration_valid:
+            # print("Consideration is not valid, move cursor")
+            self.action.consideration.execute()
+            return False
+
+        # do we need to navigate to an action?
+        # if we are on the selected character, run the appraisal:
+        if consideration_valid:
+            # print("Try to execute the appraisal")
+            self.action.appraisal.execute()
+            if self.action.appraisal.complete:
+                # print("appraisal is complete, reset action")
+                self.action = None
+            return False
+
+        return False
```

Comment:
This function only seems to return `False`. Is there a time when it would return true, and what does that mean in that case? Is the return value used by anything?

Reply:
I was intending for this to handle the sequencer lifecycle - so ill refactor and make this cleaner.

Reply:
Kinda depends how we want the sequencer to handle the node. Currently it's "move from A to B and handle any battle(s) along the way". It could also be "Move from A towards B, and terminate when battle finished".